sympy/series/sequences.py

```diff
@@ -0,0 +1,916 @@
+        """Returns a tuple of variables that are bounded"""
+        return ()
+
+    @property
+    def free_symbols(self):
+        """
+        This method returns the symbols in the object, excluding those
+        that take on a specific value (i.e. the dummy symbols).
+
+        Examples
+        ========
+
+        >>> from sympy import SeqFormula
+        >>> from sympy.abc import n, m
+        >>> SeqFormula(m*n**2, (n, 0, 5)).free_symbols
+        set([m])
+        """
+        fsyms = set().union(*[a.free_symbols for a in self.args])
+        for d in self.variables:
+            if d in fsyms:
+                fsyms.remove(d)
+        return fsyms
+
+    @cacheit
+    def coeff(self, pt):
+        """Returns the coefficient at point pt"""
+        if pt < self.start or pt > self.stop:
+            raise IndexError("Index %s out of bounds %s" %(pt, self.interval))
+        return self._eval_coeff(pt)
+
+    def _eval_coeff(self, pt):
+        raise NotImplementedError("The _eval_coeff method should be added to"
+                                  " %s to return coefficient so it is available"
+                                  " when coeff calls it."
+                                  % self.func)
+
+    def _ith_point(self, i):
+        """
+        Returns the i'th point of a sequence
+        If start point is negative infinity, point is returned from the end.
+        Assumes the first point to be indexed zero.
+
+        Examples
+        =========
+
+        >>> from sympy import oo
+        >>> from sympy.series.sequences import SeqPer
+
+        bounded
+
+        >>> SeqPer((1, 2, 3), (-10, 10))._ith_point(0)
+        -10
+        >>> SeqPer((1, 2, 3), (-10, 10))._ith_point(5)
+        -5
+
+        End is at infinity
+
+        >>> SeqPer((1, 2, 3), (0, oo))._ith_point(5)
+        5
+
+        Starts at negative infinity
+
+        >>> SeqPer((1, 2, 3), (-oo, 0))._ith_point(5)
+        -5
+        """
+        if self.start is S.NegativeInfinity:
+            initial = self.stop
+        else:
+            initial = self.start
+
+        if self.start is S.NegativeInfinity:
+            step = -1
+        else:
+            step = 1
+
+        return initial + i*step
+
+    def _add(self, other):
+        """
+        Should only be used internally
+
+        self._add(other) returns a new, term-wise added sequence if self
+        knows how to add with other, otherwise it returns ``None``.
+
+        ``other`` should only be a sequence object.
+
+        Used within :class:`SeqAdd` class
+        """
+        return None
+
+    def _mul(self, other):
+        """
+        Should only be used internally
+
+        self._mul(other) returns a new, term-wise multiplied sequence if self
+        knows how to multiply with other, otherwise it returns ``None``.
+
+        ``other`` should only be a sequence object.
+
+        Used within :class:`SeqMul` class
+        """
+        return None
+
+    def coeff_mul(self, other):
```

Comment:
Is it okay, if it's kept separate, and * knows how to multiply sequences with sequences only?